docs/source/user-guide/dataset_object.rst

```diff
@@ -3,76 +3,74 @@
-        The higher the value, the chance of column inferred as categorical is higher
-      - 0.01
-    * - max_categories
-      - int
-      - Used to infer which columns are categorical (if `cat_features` isn't explicitly passed).
-        Set maximum number of unique values in a column in order for it to be categorical.
-        The higher the value, the chance of column inferred as categorical is higher
-      - 30
-    * - max_float_categories
-      - int
-      - Same as `max_categories` but for columns of type float
-      - 5
-    * - convert_date
-      - bool
-      - Whether to convert date column if it's numeric to date
-      - True
+
+
+A Quick Introduction
+====================
+
+The Dataset Object is a simple and robust Object to store tabular data and metadata relevant to Machine Learning.
+It allows deepchecks to understand the context of the data with minimal hassle for the user.
+
+Common information used in Machine Learning that are aggregated in the Dataset:
+ - **cat_features** - Categorical features normally require some preprocessing before being passed to the model. If not specified, the categorical features are inferred automatically from the data itself. (`Inferring Features And Categorical Features`_ )
+ - **label** - The classes of the classification problem or values of a regression problem.
+ - **index_name** - This can be useful to track indexes duplicate etc. Like the other arguments, this is not mandatory.
+ - **date_name** - The date the sample was acquired (Useful for TimeSeries problems). This column is standardized using `pd.Timestamp`.
+ - **features** - Specifies the columns used by model for training. Used for defining only a subset of the columns in the data as features. If not defined, all columns that aren't explicitly specified as `label`, `date`, or `index` are considered as features.
+
+All these and more can be defined and accessed on the Dataset Object.
+This allows the user the flexibility to define these column names whichever they seem fit,
+and allows use of the these tags to standardize development around this data.
+
+
+The Basics
+==========
+
+Lets start with building a simple pd.DataFrame that we might already know as a popular data structure to store data.
+But we also have other metadata that we have other use for, and not only as features for the model.
+
+    >>> d = {"id": [1,2,3,4],
+    ...      "feature1": [0.1,0.3,0.2,0.6],
+    ...      "feature2": [4,5,6,7],
+    ...      "cat_feature": [0,0,0,1],
+    ...      "class": [1,2,1,2]}
+    ... df = pd.DataFrame(d)
+    ... dataset = Dataset(df, label="class", index_name="id", cat_features=["category"])
+
+Now we have built a Dataset that we can easily pull features and labels from:
+
+    >>> dataset.features
+    ['feature1', 'feature2', 'category']
+    >>> dataset.label_name
+    ['class']
+
+We can also get pd.DataFrames from columns that interest us e.g.:
```

Comment:
Maybe more interesting to demonstrate instead (or to state both options and show one of them) - how to show the whole dataframe with `dataset.data`